test/core/event_engine/query_extensions_test.cc

```diff
@@ -0,0 +1,93 @@
+// Copyright 2023 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include "src/core/lib/event_engine/query_extensions.h"
+
+#include <grpc/event_engine/event_engine.h>
+#include <grpc/event_engine/slice_buffer.h>
+#include <grpc/support/port_platform.h>
+
+#include <string>
+
+#include "absl/functional/any_invocable.h"
+#include "absl/status/status.h"
+#include "gtest/gtest.h"
+#include "src/core/lib/gprpp/crash.h"
+
+namespace grpc_event_engine {
+namespace experimental {
+namespace {
+
+template <int i>
+class TestExtension {
+ public:
+  TestExtension() = default;
+  ~TestExtension() = default;
+
+  static std::string EndpointExtensionName() {
```

Comment:
suggest constexpr static string

Reply:
Its not considered constexpr here due to the std::to_string(i) statement. I think it might be preferable to leave the EndpointExtensionName() function instead of a string member because static string members cannot be initialized inline.